crypto/cryptlib.c

```diff
@@ -18,7 +18,7 @@
 
 extern unsigned int OPENSSL_ia32cap_P[4];
 
-# if defined(OPENSSL_CPUID_OBJ) && !defined(OPENSSL_NO_ASM) && !defined(I386_ONLY)
+# if defined(OPENSSL_CPUID_OBJ)
```

Comment:
Why are these removed - maybe a line in the commit?

Reply:
Ok will do.

All other files use the ia32cap_P if defined(OPENSSL_CPUID_OBJ)
but when no-asm or 386 is used to configure the CPUID_OBJ is not defined.
OTOH if the OPENSSL_CPUID_OBJ is linked it it outght to be initialized no
matter what other defines say.

Reply:
I think your commit message is confusing. OPENSSL_ia32cap_P always exists on x86/x86_64. OPENSSL_CPUID_OBJ just indicates that it's defined in some assembler file and that cryptlib.c shouldn't define it. It's a global variable, and so is initialized to 0. So I think most things that check for OPENSSL_CPUID_OBJ can just remove that check.

Reply:
Maybe, but checking a OPENSSL_ia32cap_P that is not initialized is pointless.
Nevertheless **this** ifdef cannot be removed.

Reply:
I wasn't suggesting that this ifdef can be removed. (But the extern unsigned int just above it can.)

Reply:
It is likely that the header file is not included here, but that can be another PR.